### PR TITLE
Remove Global Iterators

### PR DESCRIPTION
**Pull request by skull132**
Removes the global iterators, which mechs use heavily, and replaces it with standard process() calls from SSfast_process and SSprocessing.

Need to test, but the bulk of the work is done and it compiles!

### CODE REVIEW

File: code/game/mecha/equipment/tools/medical_tools.dm [hunk 1]
```diff
@@ -379,265 +360,272 @@
...
+/obj/item/mecha_parts/mecha_equipment/tool/syringe_gun/New()
+	..()
+	flags |= NOREACT
+	syringes = new
+	known_reagents = list("inaprovaline"="Inaprovaline","anti_toxin"="Dylovene")
+	processed_reagents = new
+	create_reagents(max_volume)
+
+/obj/item/mecha_parts/mecha_equipment/tool/syringe_gun/Destroy()
+	STOP_PROCESSING(SSprocessing, src)
```
Comment (skull132) [inline code/game/mecha/equipment/tools/medical_tools.dm]: Fast process this. It ticks every 10 ds, not 20.

File: code/game/mecha/equipment/tools/medical_tools.dm [hunk 2]
```diff
@@ -379,265 +360,272 @@
...
+	var/turf/trg = get_turf(target)
+	var/obj/item/weapon/reagent_containers/syringe/S = syringes[1]
+	S.forceMove(get_turf(chassis))
+	reagents.trans_to_obj(S, min(S.volume, reagents.total_volume))
+	syringes -= S
+	S.icon = 'icons/obj/chemical.dmi'
+	S.icon_state = "syringeproj"
+	playsound(chassis, 'sound/items/syringeproj.ogg', 50, 1)
+	log_message("Launched [S] from [src], targeting [target].")
+	spawn(-1)
```
Comment (skull132) [inline code/game/mecha/equipment/tools/medical_tools.dm]: INVOKE_ASYNC this.

File: code/game/mecha/equipment/tools/tools.dm [hunk 3]
```diff
@@ -186,64 +186,64 @@
...
-			var/turf/T2 = get_step(T,turn(direction, -90))
+		var/turf/T = get_turf(target)
+		var/turf/T1 = get_step(T,turn(direction, 90))
+		var/turf/T2 = get_step(T,turn(direction, -90))
+
+		var/list/the_targets = list(T,T1,T2)
+
+		for(var/a = 1 to 5)
+			spawn(0)
+				var/obj/effect/effect/water/W = new /obj/effect/effect/water(get_turf(chassis))
```
Comment (skull132) [inline code/game/mecha/equipment/tools/tools.dm]: `INVOKE_ASYNC`

File: code/game/mecha/equipment/tools/tools.dm [hunk 4]
```diff
@@ -380,45 +378,45 @@
...
+	var/obj/effect/portal/P = new /obj/effect/portal(get_turf(target))
+	P.target = target_turf
+	P.creator = null
+	P.icon = 'icons/obj/objects.dmi'
+	P.failchance = 0
+	P.icon_state = "anom"
+	P.name = "wormhole"
+	do_after_cooldown()
+	src = null
+	spawn(rand(150,300))
```
Comment (skull132) [inline code/game/mecha/equipment/tools/tools.dm]: Callback.

File: code/game/mecha/equipment/tools/tools.dm [hunk 5]
```diff
@@ -434,66 +432,65 @@
...
+			if(!action_checks(target)) return
+			var/list/atoms = list()
+			if(isturf(target))
+				atoms = range(target,3)
+			else
+				atoms = orange(target,3)
+			for(var/atom/movable/A in atoms)
+				if(A.anchored) continue
+				spawn(0)
+					var/iter = 5-get_dist(A,target)
```
Comment (skull132) [inline code/game/mecha/equipment/tools/tools.dm]: INVOKE_ASYNC

File: code/game/mecha/equipment/tools/tools.dm [hunk 6]
```diff
@@ -594,89 +591,76 @@
...
-		if(!chassis) return
-		return "<span style=\"color:[equip_ready?"#0f0":"#f00"];\">*</span>&nbsp;[src.name] - <a href='?src=\ref[src];toggle_repairs=1'>[pr_repair_droid.active()?"Dea":"A"]ctivate</a>"
+/obj/item/mecha_parts/mecha_equipment/repair_droid/attach(obj/mecha/M as obj)
+	..()
+	droid_overlay = new(src.icon, icon_state = "repair_droid")
+	M.overlays += droid_overlay
+	return
 
+/obj/item/mecha_parts/mecha_equipment/repair_droid/destroy()
+	chassis.overlays -= droid_overlay
```
Comment (skull132) [inline code/game/mecha/equipment/tools/tools.dm]: `STOP_PROCESSING()` just in cakes.

File: code/game/mecha/equipment/tools/tools.dm [hunk 7]
```diff
@@ -772,133 +741,121 @@
...
-		if(href_list["toggle"])
-			if(pr_mech_generator.toggle())
-				set_ready_state(0)
-				log_message("Activated.")
-			else
-				set_ready_state(1)
-				log_message("Deactivated.")
-		return
+/obj/item/mecha_parts/mecha_equipment/generator/Destroy()
+	STOP_PROCESSING(SSprocessing, src)
```
Comment (skull132) [inline code/game/mecha/equipment/tools/tools.dm]: Fast process every other tick.

File: code/game/mecha/equipment/tools/tools.dm [hunk 8]
```diff
@@ -945,56 +893,56 @@
...
+				chassis.occupant_message("You lift [target] and start to load it into cargo compartment.")
+				chassis.visible_message("[chassis] lifts [target] and starts to load it into cargo compartment.")
+				set_ready_state(0)
+				chassis.use_power(energy_drain)
+				O.anchored = 1
+				var/T = chassis.loc
+				if(do_after_cooldown(target))
+					if(T == chassis.loc && src == chassis.selected)
+						cargo_holder.cargo += O
+						O.loc = chassis
```
Comment (skull132) [inline code/game/mecha/equipment/tools/tools.dm]: Forcemove.

**Comment by skull132** [comment]
Tested & confirmed working now.

File: code/game/mecha/equipment/tools/tools.dm [hunk 9]
```diff
@@ -594,89 +599,77 @@
...
+/obj/item/mecha_parts/mecha_equipment/repair_droid/Destroy()
+	STOP_PROCESSING(SSprocessing, src)
+	return ..()
 
-	get_equip_info()
-		if(!chassis) return
-		return "<span style=\"color:[equip_ready?"#0f0":"#f00"];\">*</span>&nbsp;[src.name] - <a href='?src=\ref[src];toggle_repairs=1'>[pr_repair_droid.active()?"Dea":"A"]ctivate</a>"
+/obj/item/mecha_parts/mecha_equipment/repair_droid/attach(obj/mecha/M as obj)
+	..()
+	droid_overlay = new(src.icon, icon_state = "repair_droid")
```
Comment (Lohikar) [inline code/game/mecha/equipment/tools/tools.dm]: Why is this an icon instead of an image?

File: code/game/mecha/equipment/tools/tools.dm [hunk 10]
```diff
@@ -594,89 +599,77 @@
...
+	M.overlays += droid_overlay
```
Comment (Lohikar) [inline code/game/mecha/equipment/tools/tools.dm]: This should probably be `add_overlay()`, but make sure that if there's any other overlays that they use it too.

**Pull request merged**